### PR TITLE
Fix hotbar items drag-n-drop bugs

### LINKED ISSUE #2830: Fix hotbar items dragged over iframe
Draggable items can freeze in one place if moved over cluster dashboard.

draggin over iframe

### LINKED ISSUE #4432: Jumping hotbar items onDragEnd
There are multiple mutation operations on hotbar items made and this cause jumping after dragging an item:

https://user-images.githubusercontent.com/9607060/143548766-fbfafe54-0577-4bf4-a04f-05422d4e6740.mov

### PR DESCRIPTION
**Pull request by aleksfront**
Fixes #2830 (Impossible to drag over iframe)
Fixes #4432 (Items are jumping onDragEnd)

https://user-images.githubusercontent.com/9607060/143549818-3ed3eb6b-6f23-438f-a368-e6b5c3974061.mov

### CODE REVIEW

**Comment by jweak** [comment]
Works otherwise really well, but there's some flicker when releasing the item. The item is rendered in the old place after drop and then it moves to the right place. What causes that? It seems that it's not actually regression though, since the previous version has that as well.

**Comment by aleksfront** [comment]
> Works otherwise really well, but there's some flicker when releasing the item. The item is rendered in the old place after drop and then it moves to the right place. What causes that? It seems that it's not actually regression though, since the previous version has that as well.

I don't see the item is rendered in the old place. Can you show the example?

File: src/renderer/components/hotbar/hotbar-menu.tsx [hunk 1]
```diff
@@ -54,9 +62,15 @@ export class HotbarMenu extends React.Component<Props> {
     return catalogEntityRegistry.getById(item?.entity.uid) ?? null;
   }
 
+  onDragStart() {
+    this.draggingOver = true;
+  }
+
   onDragEnd(result: DropResult) {
```
Comment (Nokel81) [inline src/renderer/components/hotbar/hotbar-menu.tsx:71]: Shouldn't this be `action`?

File: src/renderer/components/hotbar/hotbar-menu.tsx [hunk 2]
```diff
@@ -165,9 +179,9 @@ export class HotbarMenu extends React.Component<Props> {
     const hotbar = hotbarStore.getActive();
 
     return (
-      <div className={cssNames("HotbarMenu flex column", className)}>
+      <div className={cssNames("HotbarMenu flex column", { draggingOver: this.draggingOver }, className)}>
         <div className="HotbarItems flex column gaps">
-          <DragDropContext onDragEnd={this.onDragEnd.bind(this)}>
+          <DragDropContext onDragStart={this.onDragStart.bind(this)} onDragEnd={this.onDragEnd.bind(this)}>
```
Comment (Nokel81) [inline src/renderer/components/hotbar/hotbar-menu.tsx]: Can you use properties instead of `bind`?

**Review by Nokel81 — APPROVED** [review]

**Pull request merged**